pkg/lib/config/testdata/config_tests.yaml

```diff
@@ -1200,3 +1200,39 @@ config:
     signing:
       key_id: asdfasdf
       signature_method: http://www.w3.org/2001/04/xmldsig-more#rsa-sha256
+---
+name: saml-service-provider-missing-id
+error: |-
+  invalid configuration:
+  /saml/service_providers/0: required
+    map[actual:[acs_urls nameid_format] expected:[id] missing:[id]]
+  /saml/service_providers/0: required
+    map[actual:[acs_urls nameid_format] expected:[client_id] missing:[client_id]]
+config:
+  id: test
+  http:
+    public_origin: http://test
+  saml:
+    signing:
+      key_id: asdfasdf
+    service_providers:
+      - nameid_format: urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress
+        acs_urls:
+          - http://localhost:3000/acs
```

Comment:
Add two more clients here to show the validation pass case. That is
1. A SP with `id` should pass
2. A SP with `client_id` should pass

Reply:
Added two items